fiber-unix/src/scheduler.ml

```diff
@@ -128,15 +135,10 @@ end
 
 module Timer_id = Id.Make ()
 
-type detached_task =
-  { name : string option
-  ; task : unit -> unit Fiber.t
-  }
-
 type t =
   { mutable events_pending : int
   ; events : event Queue.t
-  ; mutex : Mutex.t
+  ; events_mutex : Mutex.t
   ; time_mutex : Mutex.t
   ; event_ready : Condition.t
   ; timers_available : Condition.t
```

Comment:
I have started working on the review: I have more-or-less covered new version of `Fiber` lib (with slight rewrite and comments, it's much easier to understand, and comments help a lot!) I will continue tomorrow. 

I noticed that we have several places where mutexes are interleaved with data, e.g., the snippet selected. I can think of an easy change that would make code easier to understand (by removing boilerplate-y record fields) and less error-prone (by keeping mutex associated with the data closer to the data):

```ocaml
type 'data mutexed = { v : 'data ; m : Mutex.t }
let with_mutex_locked t ~f =
    Mutex.lock t.m;
    let res = f t.v in 
    Mutex.unlock t.m;
    res 
```

Reply:
Thanks for the review. I'll add some comments.

> I noticed that we have several places where mutexes are interleaved with data, e.g., the snippet selected. I can think of an easy change that would make code easier to understand (by removing boilerplate-y record fields) and less error-prone (by keeping mutex associated with the data closer to the data):

It's worth experimenting with. Even better would be to replace things with mvars whenever possible.